### ISSUE TITLE
ETag with template digests ignoring controller namespace

### ISSUE THREAD

**Issue by fivedigit** [rails/rails#20333]
The `EtagWithTemplateDigest` concern is ignoring the fact that controllers may be namespaced while looking up the template digest to include in the ETag for a request.

Given the following controller:

```ruby
module Foo
  class BarsController < ApplicationController
    def show
      @bar = Bar.find(params[:id])

      respond_to(:html) if stale?(@bar)
    end
  end
end
```

When requesting the page, a warning appears in the output:

> Couldn't find template for digesting: bars/show

The implicit view to be rendered for this controller is `foo/bars/show`, so `EtagWithTemplateDigest` should ideally honor this convention.

**Comment by agis-** [comment]
Are you trying with 4.2 or?

**Comment by ryantownsend** [comment]
I'm also having this issue with any namespaced controller, following the default paths, e.g. `Api::V1::UsersController` would be in 'app/controllers/api/v1/users_controller.rb' with it's views in `app/views/api/v1/users`.

My Rails version: 4.2.1.

**Comment by chirag7jain** [comment]
even i am having the same issue. I am using rails 4.2.4

**Comment by chirag7jain** [comment]
@rafaelfranca has this issue been resolved

**Comment by chirag7jain** [comment]
this is not fixed.. does anyone have a workaround for this

**Comment by rafaelfranca** [comment]
Fixed by #25817
